File: src/main/java/net/snipsniper/configwindow/textpreviewwindow/SaveFormatPreviewRenderer.java
package net.snipsniper.configwindow.textpreviewwindow;

import net.snipsniper.utils.DrawUtils;
import net.snipsniper.utils.Utils;

import javax.swing.*;
import java.awt.*;

public class SaveFormatPreviewRenderer extends JPanel {
    private TextPreviewWindow textPreviewWindow;

    public static final String DEFAULT_FORMAT = "%year%-%month%-%day%__%hour%_%minute%_%second%";

    public SaveFormatPreviewRenderer(int minWidth, int minHeight) {
        Dimension min = new Dimension(minWidth, minHeight);
        setPreferredSize(min);
        setMinimumSize(min);
    }

    public void setTextPreviewWindow(TextPreviewWindow window) {
        textPreviewWindow = window;
    }

    @Override
    public void paint(Graphics g) {
        g.setColor(Color.LIGHT_GRAY);
        g.fillRect(0, 0, getWidth(), getHeight());
        g.setColor(Color.BLACK);
        String raw = textPreviewWindow.getText();
        if(raw.isEmpty())
            raw = DEFAULT_FORMAT;
        String text = Utils.Companion.constructFilename(raw, "");
        int margin = 100;
        DrawUtils.Companion.drawCenteredString(g, text, new Rectangle(0, 0, getWidth(), getHeight()), new Font("Arial", Font.BOLD, DrawUtils.Companion.pickOptimalFontSize((Graphics2D) g, text, getWidth() - margin, getHeight())));
    }

}

File: src/main/kotlin/net/snipsniper/configwindow/textpreviewwindow/SaveFormatPreviewRenderer.kt
package net.snipsniper.configwindow.textpreviewwindow

import net.snipsniper.SnipSniper
import net.snipsniper.config.ConfigHelper
import net.snipsniper.utils.DrawUtils
import net.snipsniper.utils.Utils
import java.awt.*
import javax.swing.JPanel

class SaveFormatPreviewRenderer(width: Int, height: Int) : JPanel() {
    lateinit var textPreviewWindow: TextPreviewWindow

    init {
        Dimension(width, height).also {
            preferredSize = it
            minimumSize = it
        }
    }

    companion object {
        val DEFAULT_FORMAT = SnipSniper.defaultProfileConfig.getString(ConfigHelper.PROFILE.saveFormat)
    }

    override fun paint(g: Graphics) {
        g.color = Color.LIGHT_GRAY
        g.fillRect(0, 0, width, height)
        g.color = Color.BLACK
        var raw = textPreviewWindow.text
        if(raw.isEmpty())
            raw = DEFAULT_FORMAT
        val text = Utils.constructFilename(raw, "")
        val margin = 100
        val font = Font("Arial", Font.BOLD, DrawUtils.pickOptimalFontSize(g as Graphics2D, text, width - margin, height))
        DrawUtils.drawCenteredString(g, text, Rectangle(0, 0, width, height), font)
    }
}